src/components/Breadcrumbs/Breadcrumbs.vue

```diff
@@ -0,0 +1,362 @@
+					hiddenIndices.push(currentIndex)
+					i++
+				}
+				// We only update the hidden indices if they have changed,
+				// otherwise we will run into an infinite update loop.
+				if (!this.arraysEqual(this.hiddenIndices, hiddenIndices.sort())) {
+					this.hiddenIndices = hiddenIndices
+					// Also update hiddenCrumbs here so they are in sync
+					const hiddenCrumbs = []
+					for (let jj = 0; jj < this.hiddenIndices.length; jj++) {
+						hiddenCrumbs.push(breadcrumbs[hiddenIndices[jj]])
+					}
+					this.hiddenCrumbs = hiddenCrumbs
+				}
+			}
+		},
+
+		arraysEqual(a, b) {
+			if (a.length !== b.length) return false
+			if (a === b) return true
+			if (a === null || b === null) return false
+
+			for (let i = 0; i < a.length; ++i) {
+				if (a[i] !== b[i]) return false
+			}
+			return true
+		},
+
+		getTotalWidth(breadcrumbs) {
+			return breadcrumbs.reduce((width, crumb, index) => width + this.getWidth(crumb.elm), 0)
+		},
+
+		getWidth(el) {
+			const hide = el.classList.contains('hidden')
+			el.classList.remove('hidden')
+			const w = el.offsetWidth
+			if (hide) {
+				el.classList.add('hidden')
+			}
+			return w
+		},
+		cancelEvent(e) {
+			e.stopPropagation()
+			e.preventDefault()
+			return false
+		},
+		dropOnActions(e) {
+			this.actionsOpen = false
+			return this.cancelEvent(e)
+		},
+		dragstart(e) {
+			return this.cancelEvent(e)
+		},
+		dropped(index, e) {
+			this.cancelEvent(e)
+			// If it is the last element in the path,
+			// don't do anything
+			if (index === (this.breadcrumbs.length - 1)) {
+				return
+			}
+			this.$emit('dropped', this.breadcrumbs[index].path)
+			this.actionsOpen = false
+			const crumbs = document.querySelectorAll('.crumb')
+			crumbs.forEach((f) => { f.classList.remove('over') })
+			return false
+		},
+		dragOver(e) {
+			if (e.preventDefault) {
+				e.preventDefault()
+			}
+			return false
+		},
+		dragEnter(index, e) {
+			// If it is the last element in the path,
+			// don't do anything
+			if (index === (this.breadcrumbs.length - 1)) {
+				return
+			}
+			// Get the correct element, in case we hover a child.
+			if (e.target.closest) {
+				const target = e.target.closest('.crumb')
+				if (target.classList && target.classList.contains('crumb')) {
+					const crumbs = document.querySelectorAll('.crumb')
+					crumbs.forEach((f) => { f.classList.remove('over') })
+					target.classList.add('over')
+				}
+			}
+		},
+		dragLeave(e) {
+			// Don't do anything if we leave towards a child element.
+			if (e.target.contains(e.relatedTarget)) {
+				return
+			}
+			// Get the correct element, in case we leave directly from a child.
+			if (e.target.closest) {
+				const target = e.target.closest('.crumb')
+				if (target.contains(e.relatedTarget)) {
+					return
+				}
+				if (target.classList && target.classList.contains('crumb')) {
+					target.classList.remove('over')
+				}
+			}
+		},
+	},
+	render: function(h) {
```

Comment:
Maybe not use `h` here but a nice name so it's easier to follow. Vuejs suggest usng `createElement` iirc